packages/nx/src/command-line/import/import.ts

```diff
@@ -391,76 +392,66 @@ async function createTemporaryRemote(
   await destinationGitClient.fetch(remoteName);
 }
 
-// If the user imports a project that isn't in NPM/Yarn/PNPM workspaces, then its dependencies
-// will not be installed. We should warn users and provide instructions on how to fix this.
-async function warnOnMissingWorkspacesEntry(
+/**
+ * If the user imports a project that isn't in the workspaces entry, we should add that path to the workspaces entry.
+ */
+async function handleMissingWorkspacesEntry(
   pm: PackageManager,
-  pmc: PackageManagerCommands,
-  pkgPath: string
+  pkgPath: string,
+  destinationGitClient: GitRepository
 ) {
+  let added = false;
+  try {
+    added = addPackagePathToWorkspaces(pm, workspaceRoot, pkgPath);
+  } catch {
+    return;
+  }
+  if (!added) {
+    return;
+  }
+  await destinationGitClient.amendCommit();
```

Comment:
I think this should all be under the `isWorkspacesEnabled` section right?

Reply:
changed